### PR TITLE
Task/ceremony state

### LINKED ISSUE #182: ✨ Create Tally Screen
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

Need a new page for creating a new tally.  This needs to have a name, and the choice of what ballots.  This means that the user can select all ballots or some based on device and date.

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SteveMaier-IRT**
### Issue
Fixes #182 

### Description
Getting tally to complete and create accumulated tallies and plain text tallys

### Testing
Create an election and add ballots and run a tally

### CODE REVIEW

File: bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Decryption/CiphertextDecryptionTally.cs [hunk 1]
```diff
@@ -418,9 +429,11 @@ public bool ValidateChallengeResponses()
                 challengeBallots,
                 share,
                 _challenges[guardianId]))
-            {
-                Console.WriteLine("Invalid challenge response for guardian " + guardianId);
+            {
+//                DecryptionState = DecryptionState.PendingAdminComputeProofs;
+
                 return false;
+                throw new ElectionGuardException("Invalid challenge response for guardian " + guardianId);
```
Comment (SteveMaier-IRT) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Decryption/CiphertextDecryptionTally.cs:436]: Cleaned this up to put the Console.Wrileline back in and to get rid of the throw

File: bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup/Guardian/GuardianStorageExtensions.cs [hunk 2]
```diff
@@ -17,12 +18,16 @@ public static class GuardianStorageExtensions
         GuardianPrivateRecord privateGuardianRecord,
         string keyCeremonyId,
         int numberOfGuardians,
-        int quorum)
+        int quorum,
+        Dictionary<string, ElectionPublicKey>? publicKeys = null,
+        Dictionary<string, ElectionPartialKeyBackup>? otherBackups = null)
```
Comment (SteveMaier-IRT) [inline bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup/Guardian/GuardianStorageExtensions.cs:23]: Needed to add in the publickeys and the other backups duringt he initialization.

File: bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/DecryptionMediator.cs [hunk 3]
```diff
@@ -223,6 +223,10 @@ public void AccumulateShares(string tallyId, bool skipValidation = false)
     {
         return TallyDecryptions[tallyId].CreateChallenge(skipValidation);
     }
+    public void LoadChallenge(string tallyId, string guardianId, GuardianChallenge challenge)
+    {
+        TallyDecryptions[tallyId].LoadChallenge(guardianId, challenge);
```
Comment (john-s-morgan) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/DecryptionMediator.cs:228]: I think this is the spot we should be calling the accumulator

Comment (SteveMaier-IRT) [reply bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/DecryptionMediator.cs:228]: Remember that the accumulate needs to be done once before we create these... and this only loads a single challenge so it is called multiple times.

File: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Utils/Converters/ElementModPConverter.cs [hunk 4]
```diff
@@ -10,7 +8,7 @@ public class ElementModPConverter : JsonConverter<ElementModP>
         public override ElementModP ReadJson(JsonReader reader, Type objectType, ElementModP existingValue, bool hasExistingValue, JsonSerializer serializer)
         {
             var value = (string)reader.Value;
-            return new ElementModP(value);
+            return value != null ? new ElementModP(value) : null;
```
Comment (SteveMaier-IRT) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Utils/Converters/ElementModPConverter.cs:11]: Cannot create an ElementModP from a null string

File: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Utils/Converters/ElementModQConverter.cs [hunk 5]
```diff
@@ -10,7 +8,7 @@ public class ElementModQConverter : JsonConverter<ElementModQ>
         public override ElementModQ ReadJson(JsonReader reader, Type objectType, ElementModQ existingValue, bool hasExistingValue, JsonSerializer serializer)
         {
             var value = (string)reader.Value;
-            return new ElementModQ(value);
+            return value != null ? new ElementModQ(value) : null;
```
Comment (SteveMaier-IRT) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Utils/Converters/ElementModQConverter.cs:11]: Cannot create an ElementModQ from a null string

File: src/electionguard-ui/ElectionGuard.UI/Services/TallyStateMachine.cs [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            if (currentStep is not null && await currentStep.ShouldRunStep())
+            {
+                _isRunning = true;
+                try
+                {
+                    await currentStep.RunStep();
+                }
+                catch(Exception ex)
+                {
+
```
Comment (john-s-morgan) [inline src/electionguard-ui/ElectionGuard.UI/Services/TallyStateMachine.cs:55]: TODO. make issue to bubble up errors to UI

File: src/electionguard-ui/ElectionGuard.UI/Services/TallyStateMachine.cs [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    private async Task<bool> AlwaysRun()
+    {
+        return await Task.FromResult(true);
+    }
+
+    #region Should Run
+    private async Task<bool> ShouldDecryptShares()
+    {
```
Comment (john-s-morgan) [inline src/electionguard-ui/ElectionGuard.UI/Services/TallyStateMachine.cs:78]:
```suggestion
    #region Should Run
    private async Task<bool> ShouldStartTally()
    {
        var joinedGuardians = await _tallyJoinedService.GetCountByTallyJoinedAsync(_tally.TallyId);
        return joinedGuardians >= _tally.Quorum;
    }

    private async Task<bool> AlwaysRun()
    {
        return await Task.FromResult(true);
    }

    private async Task<bool> ShouldDecryptShares()
    {
```

**Review by john-s-morgan — APPROVED** [review]

**Pull request merged**